### PR TITLE
Add a function getPodDependencies for InterpretDependency operation

### PR DESCRIPTION
**Pull request by yike21**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Introduce a new helper function accuratePodDependencies to reduce duplicate code.

**Which issue(s) this PR fixes**:
Part of #3331 

**Special notes for your reviewer**:
@Poor12 
@chaunceyjiang 

**Does this PR introduce a user-facing change?**:
It will be easy to complete `InterpretDependency` operation using lua `kube.accuratePodDependencies(desiredObj.spec.template, desiredObj.metadata.namespace)`.

```release-note
NONE
```

### CODE REVIEW

**Comment by yike21** [comment]
> /assign
> 
> Can you help post your test report?
Ok, I will post it soon! and I add a test case in ‘pkg/resourceinterpreter/configurableinterpreter/luavm/lua_test.go’.

File: pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/defaultinterpreter"
```
Comment (chaunceyjiang) [inline pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go:8]: It's easy to cause `import cycle` like this, so I suggest moving the functions to another place.

File: pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ func KubeLoader(ls *lua.LState) int {
 var kubeFuncs = map[string]lua.LGFunction{
 	"resourceAdd":             resourceAdd,
 	"accuratePodRequirements": accuratePodRequirements,
+	"accuratePodDependencies": accuratePodDependencies,
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go:40]: I don't understand why it's named accurateXXX,  which seems to be more appropriate to use getXXX.

File: pkg/resourceinterpreter/configurableinterpreter/luavm/lua_test.go [hunk 3]
```diff
@@ -551,6 +551,16 @@ func TestGetDeployPodDependencies(t *testing.T) {
 				    end`,
 			want: expect,
 		},
+		{
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/configurableinterpreter/luavm/lua_test.go:554]: Can you help add a test case for empty namespaces?

File: pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go [hunk 4]
```diff
@@ -91,3 +92,34 @@ func checkResourceQuantity(ls *lua.LState, n int) resource.Quantity {
 		return resource.Quantity{}
 	}
 }
+
+func accuratePodDependencies(ls *lua.LState) int {
+	n := ls.GetTop()
+	if n != 2 {
+		ls.RaiseError("getPodRequirements only accepts two argument")
```
Comment (jwcesign) [inline pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go:99]: getPodDependencies?

File: pkg/util/helper/luakube.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+		},
+	}
+
+	pod.Spec = *template.Spec.DeepCopy()
+	return pod
+}
+
+// GetDependenciesFromPodTemplate returns DependentObjectReferences according to podObj.
+// It is from `pkg\resourceinterpreter\defaultinterpreter\dependencies.go`.
+func GetDependenciesFromPodTemplate(podObj *corev1.Pod) ([]configv1alpha1.DependentObjectReference, error) {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/luakube.go:26]: You can directly use the `GetDependenciesFromPodTemplate` method in the `pkg\resourceinterpreter\defaultinterpreter\dependencies.go` file and set it as the export method.

File: pkg/util/helper/luakube.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
+package helper
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/luakube.go:1]: We can consider moving the directory to `pkg/util/interpreter`.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~

File: pkg/util/helper/luakube.go (hunk 6 above)
Comment (jwcesign) [inline pkg/util/helper/luakube.go:1]: how about renaming to luainterpret.go?

File: pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go (hunk 2 above)
Comment (yike21) [reply pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go:40]: It sounds good! It's named `accurateXXX` just for tidy. `getXXX` is more appropriate, I will fix it soon.

File: pkg/resourceinterpreter/configurableinterpreter/luavm/lua_test.go (hunk 3 above)
Comment (yike21) [reply pkg/resourceinterpreter/configurableinterpreter/luavm/lua_test.go:554]: Ok, I will do it soon!

File: pkg/util/helper/luakube.go (hunk 5 above)
Comment (chaunceyjiang) [reply pkg/util/helper/luakube.go:26]: Perhaps we can move this function to the `pkg/util/helper/pod.go` file.

File: pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go (hunk 4 above)
Comment (yike21) [reply pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go:99]: Thanks! I will fix it.

File: pkg/util/helper/luakube.go (hunk 5 above)
Comment (XiShanYongYe-Chang) [reply pkg/util/helper/luakube.go:26]: +1

**Comment by yike21** [comment]
This is  test report under `pkg/resourceinterpreter/customized/declarative/luavm` using `jstemmer/go-junit-report` :
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites tests="37">
	<testsuite name="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" tests="37" failures="0" errors="0" id="0" hostname="Tide" time="0.000" timestamp="2023-03-31T08:13:12+08:00">
		<testcase name="TestConvertLuaResultInto" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/not_a_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/empty_table_into_struct" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/empty_table_into_slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/non-empty_table_into_slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/table_with_empty_table_into_slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/struct_is_not_empty,_and_convert_successfully" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetReplicas" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetReplicas/Test_GetReplica_with_kube.accuratePodRequirements" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetReplicas/Test_GetReplica_with_kube.resourceAdd" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestReviseDeploymentReplica" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestReviseDeploymentReplica/Test_ReviseDeploymentReplica" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000">
			<system-out><![CDATA[    lua_test.go:247: Success Test]]></system-out>
		</testcase>
		<testcase name="TestReviseDeploymentReplica/revise_deployment_replica" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000">
			<system-out><![CDATA[    lua_test.go:247: Success Test]]></system-out>
		</testcase>
		<testcase name="TestAggregateDeploymentStatus" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestHealthDeploymentStatus" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestRetainDeployment" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestRetainDeployment/Test_RetainDeployment1" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestRetainDeployment/Test_RetainDeployment2" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestStatusReflection" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestStatusReflection/Test_StatusReflection" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetDeployPodDependencies" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/nil" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/nil_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/int_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/int" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/uint" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/float" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/bool" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/string" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/json_number" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/slice_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/struct" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/struct_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/[]interface{}" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/map[string]interface{}" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
	</testsuite>
</testsuites>
```
This is  test result under `pkg/util/helper` :
```shell
go.exe test -timeout 30s -run ^(TestGetPodCondition|TestGetPodConditionFromList|TestGeneratePodFromTemplateAndNamespace|TestGetSecretNames|TestGetConfigMapNames|TestGetPVCNames|TestGetDependenciesFromPodTemplate|Test_getServiceAccountNames)$ github.com/karmada-io/karmada/pkg/util/helper

=== RUN   TestGetPodCondition
=== RUN   TestGetPodCondition/empty_status
=== RUN   TestGetPodCondition/empty_condition
=== RUN   TestGetPodCondition/doesn't_have_objective_condition
=== RUN   TestGetPodCondition/have_objective_condition
--- PASS: TestGetPodCondition (0.00s)
    --- PASS: TestGetPodCondition/empty_status (0.00s)
    --- PASS: TestGetPodCondition/empty_condition (0.00s)
    --- PASS: TestGetPodCondition/doesn't_have_objective_condition (0.00s)
    --- PASS: TestGetPodCondition/have_objective_condition (0.00s)
=== RUN   TestGetPodConditionFromList
=== RUN   TestGetPodConditionFromList/empty_condition
=== RUN   TestGetPodConditionFromList/doesn't_have_objective_condition
=== RUN   TestGetPodConditionFromList/have_objective_condition
=== RUN   TestGetPodConditionFromList/PodReady_condition
--- PASS: TestGetPodConditionFromList (0.00s)
    --- PASS: TestGetPodConditionFromList/empty_condition (0.00s)
    --- PASS: TestGetPodConditionFromList/doesn't_have_objective_condition (0.00s)
    --- PASS: TestGetPodConditionFromList/have_objective_condition (0.00s)
    --- PASS: TestGetPodConditionFromList/PodReady_condition (0.00s)
=== RUN   TestGeneratePodFromTemplateAndNamespace
=== RUN   TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency
=== PAUSE TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency
=== RUN   TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace
=== PAUSE TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace
=== CONT  TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency
=== CONT  TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace
--- PASS: TestGeneratePodFromTemplateAndNamespace (0.00s)
    --- PASS: TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency (0.00s)
    --- PASS: TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace (0.00s)
=== RUN   TestGetSecretNames
=== RUN   TestGetSecretNames/get_secret_names_from_pod_without_secrets
=== PAUSE TestGetSecretNames/get_secret_names_from_pod_without_secrets
=== RUN   TestGetSecretNames/get_secret_names_from_pod_with_secrets
=== PAUSE TestGetSecretNames/get_secret_names_from_pod_with_secrets
=== CONT  TestGetSecretNames/get_secret_names_from_pod_without_secrets
=== CONT  TestGetSecretNames/get_secret_names_from_pod_with_secrets
--- PASS: TestGetSecretNames (0.00s)
    --- PASS: TestGetSecretNames/get_secret_names_from_pod_without_secrets (0.00s)
    --- PASS: TestGetSecretNames/get_secret_names_from_pod_with_secrets (0.00s)
=== RUN   TestGetConfigMapNames
=== RUN   TestGetConfigMapNames/get_configMap_names_from_pod_without_configMaps
=== PAUSE TestGetConfigMapNames/get_configMap_names_from_pod_without_configMaps
=== RUN   TestGetConfigMapNames/get_configMap_names_from_pod_with_configMaps
=== PAUSE TestGetConfigMapNames/get_configMap_names_from_pod_with_configMaps
=== CONT  TestGetConfigMapNames/get_configMap_names_from_pod_without_configMaps
=== CONT  TestGetConfigMapNames/get_configMap_names_from_pod_with_configMaps
--- PASS: TestGetConfigMapNames (0.00s)
    --- PASS: TestGetConfigMapNames/get_configMap_names_from_pod_without_configMaps (0.00s)
    --- PASS: TestGetConfigMapNames/get_configMap_names_from_pod_with_configMaps (0.00s)
=== RUN   TestGetPVCNames
=== RUN   TestGetPVCNames/get_pvc_names_from_pod_without_pvcs
=== PAUSE TestGetPVCNames/get_pvc_names_from_pod_without_pvcs
=== RUN   TestGetPVCNames/get_pvc_names_from_pod_with_pvcs
=== PAUSE TestGetPVCNames/get_pvc_names_from_pod_with_pvcs
=== CONT  TestGetPVCNames/get_pvc_names_from_pod_without_pvcs
=== CONT  TestGetPVCNames/get_pvc_names_from_pod_with_pvcs
--- PASS: TestGetPVCNames (0.00s)
    --- PASS: TestGetPVCNames/get_pvc_names_from_pod_without_pvcs (0.00s)
    --- PASS: TestGetPVCNames/get_pvc_names_from_pod_with_pvcs (0.00s)
=== RUN   TestGetDependenciesFromPodTemplate
=== RUN   TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_without_dependencies
=== PAUSE TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_without_dependencies
=== RUN   TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_with_dependencies
=== PAUSE TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_with_dependencies
=== CONT  TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_without_dependencies
=== CONT  TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_with_dependencies
--- PASS: TestGetDependenciesFromPodTemplate (0.00s)
    --- PASS: TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_without_dependencies (0.00s)
    --- PASS: TestGetDependenciesFromPodTemplate/get_dependencies_from_PodTemplate_with_dependencies (0.00s)
=== RUN   Test_getServiceAccountNames
=== RUN   Test_getServiceAccountNames/get_ServiceAccountName_from_pod_
=== PAUSE Test_getServiceAccountNames/get_ServiceAccountName_from_pod_
=== RUN   Test_getServiceAccountNames/get_default_ServiceAccountName_from_pod_
=== PAUSE Test_getServiceAccountNames/get_default_ServiceAccountName_from_pod_
=== CONT  Test_getServiceAccountNames/get_ServiceAccountName_from_pod_
=== CONT  Test_getServiceAccountNames/get_default_ServiceAccountName_from_pod_
--- PASS: Test_getServiceAccountNames (0.00s)
    --- PASS: Test_getServiceAccountNames/get_ServiceAccountName_from_pod_ (0.00s)
    --- PASS: Test_getServiceAccountNames/get_default_ServiceAccountName_from_pod_ (0.00s)
PASS
ok      github.com/karmada-io/karmada/pkg/util/helper   0.153s

> 测试运行完成时间: 2023/3/31 08:24:44 <
```

File: pkg/util/helper/pod.go [hunk 7]
```diff
@@ -26,3 +31,103 @@ func GetPodConditionFromList(conditions []corev1.PodCondition, conditionType cor
 	}
 	return -1, nil
 }
+
+// GeneratePodFromTemplate generate a simple pod object from the given podTemplate and namespace, then
```
Comment (XiShanYongYe-Chang) [inline pkg/util/helper/pod.go:35]: GeneratePodFromTemplate -> GeneratePodFromTemplateAndNamespace

**Review by RainbowMango — COMMENTED** [review]
Done

**Comment by yike21** [comment]
This is  test report under `pkg/resourceinterpreter/customized/declarative/luavm` using `jstemmer/go-junit-report` :
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites tests="37">
	<testsuite name="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" tests="37" failures="0" errors="0" id="0" hostname="debian" time="0.043" timestamp="2023-03-31T23:29:50+08:00">
		<testcase name="TestConvertLuaResultInto" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/not_a_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/empty_table_into_struct" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/empty_table_into_slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/non-empty_table_into_slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/table_with_empty_table_into_slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestConvertLuaResultInto/struct_is_not_empty,_and_convert_successfully" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetReplicas" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.010"></testcase>
		<testcase name="TestGetReplicas/Test_GetReplica_with_kube.accuratePodRequirements" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetReplicas/Test_GetReplica_with_kube.resourceAdd" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestReviseDeploymentReplica" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestReviseDeploymentReplica/Test_ReviseDeploymentReplica" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000">
			<system-out><![CDATA[    lua_test.go:247: Success Test]]></system-out>
		</testcase>
		<testcase name="TestReviseDeploymentReplica/revise_deployment_replica" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000">
			<system-out><![CDATA[    lua_test.go:247: Success Test]]></system-out>
		</testcase>
		<testcase name="TestAggregateDeploymentStatus" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestHealthDeploymentStatus" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestRetainDeployment" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestRetainDeployment/Test_RetainDeployment1" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestRetainDeployment/Test_RetainDeployment2" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestStatusReflection" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestStatusReflection/Test_StatusReflection" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="TestGetDeployPodDependencies" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/nil" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/nil_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/int_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/int" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/uint" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/float" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/bool" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/string" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/json_number" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/slice" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/slice_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/struct" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/struct_pointer" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/[]interface{}" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
		<testcase name="Test_decodeValue/map[string]interface{}" classname="github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm" time="0.000"></testcase>
	</testsuite>
</testsuites>
```
And this is test result under pkg/util/helper:
```shell
go test -timeout 30s -run ^(TestGetPodCondition|TestGetPodConditionFromList|TestGeneratePodFromTemplateAndNamespace|TestGetSecretNames)$ github.com/karmada-io/karmada/pkg/util/helper

=== RUN   TestGetPodCondition
=== RUN   TestGetPodCondition/empty_status
=== RUN   TestGetPodCondition/empty_condition
=== RUN   TestGetPodCondition/doesn't_have_objective_condition
=== RUN   TestGetPodCondition/have_objective_condition
--- PASS: TestGetPodCondition (0.00s)
    --- PASS: TestGetPodCondition/empty_status (0.00s)
    --- PASS: TestGetPodCondition/empty_condition (0.00s)
    --- PASS: TestGetPodCondition/doesn't_have_objective_condition (0.00s)
    --- PASS: TestGetPodCondition/have_objective_condition (0.00s)
=== RUN   TestGetPodConditionFromList
=== RUN   TestGetPodConditionFromList/empty_condition
=== RUN   TestGetPodConditionFromList/doesn't_have_objective_condition
=== RUN   TestGetPodConditionFromList/have_objective_condition
=== RUN   TestGetPodConditionFromList/PodReady_condition
--- PASS: TestGetPodConditionFromList (0.00s)
    --- PASS: TestGetPodConditionFromList/empty_condition (0.00s)
    --- PASS: TestGetPodConditionFromList/doesn't_have_objective_condition (0.00s)
    --- PASS: TestGetPodConditionFromList/have_objective_condition (0.00s)
    --- PASS: TestGetPodConditionFromList/PodReady_condition (0.00s)
=== RUN   TestGeneratePodFromTemplateAndNamespace
=== RUN   TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency
=== PAUSE TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency
=== RUN   TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace
=== PAUSE TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace
=== CONT  TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency
=== CONT  TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace
--- PASS: TestGeneratePodFromTemplateAndNamespace (0.00s)
    --- PASS: TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace_without_dependency (0.00s)
    --- PASS: TestGeneratePodFromTemplateAndNamespace/generate_a_simple_pod_from_template_and_namespace (0.00s)
=== RUN   TestGetSecretNames
=== RUN   TestGetSecretNames/get_secret_names_from_pod_without_secrets
=== PAUSE TestGetSecretNames/get_secret_names_from_pod_without_secrets
=== RUN   TestGetSecretNames/get_secret_names_from_pod_with_secrets
=== PAUSE TestGetSecretNames/get_secret_names_from_pod_with_secrets
=== CONT  TestGetSecretNames/get_secret_names_from_pod_without_secrets
=== CONT  TestGetSecretNames/get_secret_names_from_pod_with_secrets
--- PASS: TestGetSecretNames (0.00s)
    --- PASS: TestGetSecretNames/get_secret_names_from_pod_without_secrets (0.00s)
    --- PASS: TestGetSecretNames/get_secret_names_from_pod_with_secrets (0.00s)
PASS
ok      github.com/karmada-io/karmada/pkg/util/helper   0.021s


> 测试运行完成时间: 2023/3/31 23:39:02 <
```
The difference is [To fix CI/lint](https://github.com/karmada-io/karmada/compare/7468091cfbc95dcfec296abd70a2c8726e9d8681..df4d4a0069fab6883ddb0af54d57880e4ffc40dd) and some modifications in [pkg/resourceinterpreter/customized/declarative/luavm/kube.go](https://github.com/karmada-io/karmada/compare/df4d4a0069fab6883ddb0af54d57880e4ffc40dd..dba6602d1bb97da9739ddf8b112308402cd7a611) according to suggestions, then I rebase the latest commit. 
I am sorry to trouble you so many times, please help to review and run the CI. Thank you very much! @Poor12 @XiShanYongYe-Chang @RainbowMango @ikaven1024

File: pkg/resourceinterpreter/customized/declarative/luavm/kube.go [hunk 8]
```diff
@@ -37,6 +37,7 @@ func KubeLoader(ls *lua.LState) int {
 var kubeFuncs = map[string]lua.LGFunction{
 	"resourceAdd":             resourceAdd,
 	"accuratePodRequirements": accuratePodRequirements,
+	"getPodDependencies":      getPodDependencies,
```
[the remark this answers is not in the document]
Comment (XiShanYongYe-Chang) [reply pkg/resourceinterpreter/customized/declarative/luavm/kube.go:43]: There's no documentation yet, so I've open an issue to track this task: https://github.com/karmada-io/website/issues/343

**Review by XiShanYongYe-Chang — COMMENTED** [review]
LGTM

Ask @RainbowMango to help run the CI again.

**Comment by Poor12** [comment]
LGTM

File: pkg/util/helper/pod.go [hunk 9]
```diff
@@ -26,3 +31,103 @@ func GetPodConditionFromList(conditions []corev1.PodCondition, conditionType cor
 	}
 	return -1, nil
 }
+
+// GeneratePodFromTemplateAndNamespace generate a simple pod object from the given podTemplate and namespace, then
```
Comment (RainbowMango) [inline pkg/util/helper/pod.go:35]:
```suggestion
// GeneratePodFromTemplateAndNamespace generates a simple pod object from the given podTemplate and namespace, then
```

**Comment by RainbowMango** [comment]
@Poor12 We don't need the release-notes for this improvement, right?

**Comment by Poor12** [comment]
> @Poor12 We don't need the release-notes for this improvement, right?

I think don't need. It's more like adding a helper func in kube library.

**Review by RainbowMango — APPROVED** [review]
/approve
